src/main/java/ru/itis/security/providers/JwtAuthenticationProvider.java

```diff
@@ -0,0 +1,43 @@
+package ru.itis.security.providers;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.security.authentication.AuthenticationProvider;
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.stereotype.Component;
+import ru.itis.security.auth.JwtAuthentication;
+import ru.itis.security.details.UserDetailsImpl;
+
+@Component
+public class JwtAuthenticationProvider implements AuthenticationProvider {
```

Comment:
I think it would be easier to implement JWT Authentication by yourself instead of using Spring Security.

Reply:
why?